app/controllers/admin/settings/project_life_cycle_step_definitions_controller.rb

```diff
@@ -0,0 +1,138 @@
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Admin::Settings
+  class ProjectLifeCycleStepDefinitionsController < ::Admin::SettingsController
+    menu_item :project_life_cycle_step_definitions_settings
+
+    helper_method :allowed_to_customize_life_cycle?
+
+    before_action :check_feature_flag
+    before_action :require_enterprise_token, except: %i[index]
+
+    before_action :find_definitions, only: %i[index]
+    before_action :find_definition, only: %i[edit update destroy move drop]
+
+    def index; end
+
+    def new_stage
+      @definition = Project::StageDefinition.new
+
+      render :form
+    end
+
+    def new_gate
+      @definition = Project::GateDefinition.new
+
+      render :form
+    end
+
+    def edit
+      render :form
+    end
+
+    def create
+      @definition = Project::LifeCycleStepDefinition.new(definition_params)
+
+      if @definition.save
+        flash[:notice] = I18n.t(:notice_successful_create)
+        redirect_to action: :index, status: :see_other
+      else
+        render :form, status: :unprocessable_entity
+      end
+    end
+
+    def update
+      if @definition.update(definition_params)
+        flash[:notice] = I18n.t(:notice_successful_update)
+        redirect_to action: :index, status: :see_other
+      else
+        render :form, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      if @definition.destroy
+        flash[:notice] = I18n.t(:notice_successful_delete)
+      else
+        # TODO: handle better
+        flash[:error] = I18n.t(:notice_bad_request)
+      end
+
+      redirect_to action: :index, status: :see_other
+    end
+
+    def move
+      if @definition.update(params.permit(:move_to))
+        flash[:notice] = I18n.t(:notice_successful_update)
+      else
+        # TODO: handle better
+        flash[:error] = I18n.t(:notice_bad_request)
+      end
+
+      redirect_to action: :index, status: :see_other
+    end
+
+    def drop
+      if @definition.update(params.permit(:position))
+        flash[:notice] = I18n.t(:notice_successful_update)
+      else
+        # TODO: handle better
+        flash[:error] = I18n.t(:notice_bad_request)
+      end
+
+      redirect_to action: :index, status: :see_other
```

Comment:
This redirect leads to problems since the drop action is called with a turbo stream request which results in the index action also being called with `Accept: text/vnd.turbo-stream.html` on the redirected request. 

It might be that the problem simply vanishes by not doing a redirect at all. The only reason for that redirect I currently see is to have the flash message displayed. And this opens up the question to ask: Do we actually want those flash messages to be displayed when drag&drop is used? 

It seems that on other pages where we do drag&drop, no flash is displayed. I don't know if that has been explicitly agreed upon as a UX pattern, @HDinger. Other interactions like the ordering via menu item and also using the toggle buttons on e.g. project attributes does not trigger a flash message. So it seems to me that we differentiate between different kinds of interactions. 

If those smaller ones don't need to have a flash message, then not redirecting here would be fine. What to do in an error case might be a different story.

Reply:
This was also already changed yesterday, but not pushed